### PR TITLE
Backport 2.x: Disable debugging features in baremetal

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
The baremetal configuration is, among other things, our default reference point for code size measurements. So disable debugging features that would not be enabled in production where code size is limited. In particular, this shrinks the core SSL modules by about half. Keep debugging features that are solely in their own modules (MBEDTLS_ERROR_C, MBEDTLS_VERSION_FEATURES) since it's easy to filter them out.

Named configurations are not documented interfaces, so we don't announce this in the changelog and we do it in LTS branches as well.

Backport of https://github.com/ARMmbed/mbedtls/pull/4428

### CODE REVIEW

File: scripts/config.py [hunk 1]
```diff
@@ -235,11 +235,15 @@ def full_adapter(name, active, section):
 
 # The baremetal configuration excludes options that require a library or
 # operating system feature that is typically not present on bare metal
-# systems. Features that are excluded from "full" won't be in "baremetal"
-# either (unless explicitly turned on in baremetal_adapter) so they don't
-# need to be repeated here.
+# systems. It also excludes debugging features that would typically be omitted
+# where code size is at a premium and that do not have a module of their own,
```
Comment (mstarzyk-mobica) [inline scripts/config.py:239]: "where code size is at a premium " or maybe "when code size is at a premium" ?

**Review by mstarzyk-mobica — APPROVED** [review]
lgtm, minor comment only.

File: scripts/config.py (hunk 1 above)
Comment (gilles-peskine-arm) [reply scripts/config.py]: I'm not sure what you're suggesting here, but this sentence was a bit hard to read so I've rewritten it.

**Review by mstarzyk-mobica — DISMISSED** [review]

**Comment by mstarzyk-mobica** [comment]
Windows build is failing

**Comment by gilles-peskine-arm** [comment]
The Windows build failure on Travis is a transient infrastructure error. Given that the previous build passed and the only difference is in a comment in a non-Windows-specific file, we can ignore this failure, we don't need a re-test.

File: scripts/config.py [hunk 2]
```diff
@@ -235,11 +235,17 @@ def full_adapter(name, active, section):
 
 # The baremetal configuration excludes options that require a library or
 # operating system feature that is typically not present on bare metal
-# systems. Features that are excluded from "full" won't be in "baremetal"
-# either (unless explicitly turned on in baremetal_adapter) so they don't
-# need to be repeated here.
+# systems.
+# It also excludes debugging features that increase the code size of other
+# modules sice they would typically be disabled in production in environments
```
Comment (ronald-cron-arm) [inline scripts/config.py:240]:
```suggestion
# modules since they would typically be disabled in production in environments
```

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]
There is a typo. Otherwise fine by me.

File: scripts/config.py (hunk 2 above)
Comment (gilles-peskine-arm) [reply scripts/config.py]: I amended the last commit to fix the typo.

**Review by ronald-cron-arm — APPROVED** [review]

**Comment by ronald-cron-arm** [comment]
@mstarzyk-mobica ok with you?

**Review by ronald-cron-arm — APPROVED** [review]

**Review by mstarzyk-mobica — APPROVED** [review]

**Comment by mpg** [comment]
Since the parent PR requires work, reflect this here.

**Comment by gilles-peskine-arm** [comment]
As discussed in https://github.com/ARMmbed/mbedtls/pull/4428, this approach is not good and will be superseded by a backport of https://github.com/ARMmbed/mbedtls/pull/4907.

**Pull request closed without merging**